assertions/src/main/java/io/camunda/zeebe/process/test/assertions/BpmnAssert.java

```diff
@@ -8,18 +8,32 @@
 import io.camunda.zeebe.process.test.filters.RecordStream;
 import io.camunda.zeebe.process.test.inspections.model.InspectedProcessInstance;
 
+/** This class manages all the entry points for the specific assertions. */
 public abstract class BpmnAssert {
 
   static ThreadLocal<RecordStream> recordStream = new ThreadLocal<>();
 
+  /**
+   * Initializes a {@link RecordStream}. The {@link RecordStream} will be stored in a {@link
+   * ThreadLocal} and thus will only be accessible in the current thread.
+   *
+   * @param recordStream the {@link RecordStream}
+   */
   public static void initRecordStream(final RecordStream recordStream) {
     BpmnAssert.recordStream.set(recordStream);
   }
```

Comment:
Preferably never, because it is taken care of by the test extensions.

In reality there is one case it could happen. If the user for some reason start multiple threads in his tests it needs to initialise a record stream for each thread., since it it is stored in a `ThreadLocal`